### PR TITLE
fix: regex validate plugin names on plugin add cmd

### PR DESCRIPTION
**Pull request by jthegedus**
# Summary

Apply regex validation to plugin names when `asdf plugin add` is used.

Regex is `^[a-zA-Z0-9_-]+$` and applied via `grep --extended-regexp`.

Fixes: #789 

## Other Information

`^[a-zA-Z0-9-]$` was the original discussion. I have added `_` as well as the regex will then cover all popular naming conventions (camelCase, snake_case, kebab-case, PascalCase and UPPER_SNAKE).

We may restrict this regex further in future because of platform independent handling of filesystem paths with varying cases as discussed in https://github.com/asdf-vm/asdf-plugins/issues/28

### CODE REVIEW

File: test/test_helpers.bash [hunk 1]
```diff
@@ -46,7 +46,7 @@ install_mock_broken_plugin() {
 install_mock_plugin_repo() {
   local plugin_name=$1
   local location="${BASE_DIR}/repo-${plugin_name}"
-  cp -r "$BATS_TEST_DIRNAME/fixtures/dummy_plugin" "${location}"
```
Comment (jthegedus) [inline test/test_helpers.bash:49]: The `dummy` plugin was hardcoded into this helper function. It needed to be changed to test a new plugin with different name and subsequently broke the `plugin update` tests for updating to the remote default branch.

Comment (jthegedus) [reply test/test_helpers.bash]: nvm, been staring at screens too long

**Comment by Stratus3D** [comment]
Is `grep --extended-regexp` supported on all platforms?

**Comment by jthegedus** [comment]
I am not sure of the portability of `grep --extended-regexp` (`grep -E`, I prefer the clearer extended flag names), but we're already using it in a number of places, so this change doesn't increase the surface area of tools or flags used.

I think we just wait until someone flags `grep -E` as being an issue for their platform before attempting to remove the dependence.

**Comment by Stratus3D** [comment]
@jthegedus sounds good. I didn't realize core was already using `grep -E`.

**Pull request merged**